### PR TITLE
[iota-analytics-indexer] Add README.md

### LINKED ISSUE #3976: [iota-analytics-indexer]: Review application use cases and collected metrics
## Description

The goal of this task is to review the `iota-analytics-indexer` crate to investigate the following questions

1. What is its use?
2. What metrics are collected in the application?
3. How does it compare to the current analytics solution in `iota-indexer`?

The findings should be reported in a `README.md` file in the crate.

### PR DESCRIPTION
**Pull request by tomxey**
# Description of change

Add README.md for iota-analytics-indexer

## Links to any relevant issues

#3976

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Documentation Fix

## How the change has been tested

N.A.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by kodemartin** [comment]
Had a first quick read:

* I don't get what metrics are collected though? How does it compare to the `indexer-analytical-worker`?
* Could you elaborate on the `BigQuery/Snowflake` schemas used?
* What is `BigQuery` and `Snowflake` in the first place? It would be helpful to write a few things and provide references links for digging deeper.

**Comment by tomxey** [comment]
> Had a first quick read:
> 
>     * I don't get what metrics are collected though? How does it compare to the `indexer-analytical-worker`?
> 
>     * Could you elaborate on the `BigQuery/Snowflake` schemas used?
> 
>     * What is `BigQuery` and `Snowflake` in the first place? It would be helpful to write a few things and provide references links for digging deeper.

@kodemartin Pushed changes that are addressing those remarks.

File: crates/iota-analytics-indexer/README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+### iota-indexer
+
+Currently iota-indexer is computing and storing analytical metrics about:
+ - network statistics (amount of transactions, transactions per second)
+ - (active) addresses (transactions senders/recipients)
+ - move calls
+
+ Those metrics are computed by a separate analytical worker instance of the indexer, but it uses the main DB as the main indexer instance.
+
+It seems that some of the values stored in main indexer tables by iota-indexer's `fullnode_sync_worker` are only stored there for analytical purposes (move calls, tx recipients) and could potentially be not processed/stored if not for analytical reasons.
```
Comment (kodemartin) [inline crates/iota-analytics-indexer/README.md]: Nit: Phrasing
```suggestion
It seems that some of the values stored in main indexer tables by iota-indexer's `fullnode_sync_worker` are only stored there for analytical purposes (move calls, tx recipients) and could potentially be excluded from further processing if it were not for analytical reasons. 
```

File: crates/iota-analytics-indexer/README.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+ Those metrics are computed by a separate analytical worker instance of the indexer, but it uses the main DB as the main indexer instance.
+
+It seems that some of the values stored in main indexer tables by iota-indexer's `fullnode_sync_worker` are only stored there for analytical purposes (move calls, tx recipients) and could potentially be not processed/stored if not for analytical reasons. 
+
+### iota-analytics-indexer
+
+The `iota-analytics-indexer` is not computing any analytical metrics directly.
+It is only exporting data for further processing via external tools (BigQuery/SnowFlake).
+Functionality from `iota-indexer` that is not required to serve user JSON RPC/GraphQL requests could potentially be moved away from `iota-indexer` and served by some other tool based on data exported by the `iota-analytics-indexer`.
```
Comment (kodemartin) [inline crates/iota-analytics-indexer/README.md]: Nit: Phrasing (if I get it right)
```suggestion
On this premise, the functionality  in `iota-indexer` that is currently used for extracting analytics (and thus unrelated to the JSON-RPC/GraphQL service)  could be moved out and delegated to another tool that processes data exported by `iota-analytics-indexer`.
```

File: crates/iota-analytics-indexer/README.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+The sync logic in `iota-indexer` could be simplified a bit then to store only data that is needed to serve user requests.
```
Comment (kodemartin) [inline crates/iota-analytics-indexer/README.md]: Nit:
```suggestion
Then the sync logic in `iota-indexer` could be simplified as well to store only data that is needed for the purposes of the RPC APIs.
```

File: crates/iota-analytics-indexer/README.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+ - CHECKPOINT
+ - EVENT
+ - MOVE_CALL
+ - OBJECT
+ - MOVE_PACKAGE
+ - TRANSACTION_OBJECT - input and output objects for given transactions
+ - TRANSACTION
+
+
+> Note: The following rust structs currently do not have DB schemas prepared:
```
Comment (kodemartin) [inline crates/iota-analytics-indexer/README.md]: Nit: For further emphasis (see also [GH alerts](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts))
```suggestion
> [!NOTE]
> The following rust structs currently do not have DB schemas prepared:
```

**Review by kodemartin — APPROVED** [review]
Looks good @tomxey ! Left a few optional suggestions.

Also note that the diagram does not look very clear on dark mode (there are some hatch lines overlapping with text for example).

**Review by sergiupopescu199 — APPROVED** [review]
LGTM ✨

File: crates/iota-analytics-indexer/README.md (hunk 1 above)
Comment (tomxey) [reply crates/iota-analytics-indexer/README.md]: done

File: crates/iota-analytics-indexer/README.md (hunk 2 above)
Comment (tomxey) [reply crates/iota-analytics-indexer/README.md]: Done

File: crates/iota-analytics-indexer/README.md (hunk 3 above)
Comment (tomxey) [reply crates/iota-analytics-indexer/README.md]: Done

File: crates/iota-analytics-indexer/README.md (hunk 4 above)
Comment (tomxey) [reply crates/iota-analytics-indexer/README.md]: Thanks for the tip. Done.

**Comment by tomxey** [comment]
> Also note that the diagram does not look very clear on dark mode (there are some hatch lines overlapping with text for example).

@kodemartin Changed styling of the diagram. Now it should be readable both in dark and light modes.

**Pull request merged**